cpp/src/arrow/dataset/file_base.h

```diff
@@ -350,23 +351,25 @@ class ARROW_DS_EXPORT FileWriter {
 /// \brief Options for writing a dataset.
 struct ARROW_DS_EXPORT FileSystemDatasetWriteOptions {
   /// Options for individual fragment writing.
-  std::shared_ptr<FileWriteOptions> file_write_options;
+  std::shared_ptr<FileWriteOptions> file_write_options =
+      CsvFileFormat().DefaultWriteOptions();
```

Comment:
default to ArrowFileFormat, since CSV is optional? (Though I guess, maybe you can't build datasets without CSV.)

Reply:
I think parquet is probably the best default format for writing datasets as it is generally going to be friendlier on the disk size.  Although I could be convinced that IPC is better since, as you point out, parquet is also an optional component.  I very much agree it should not be CSV.

Reply:
It seems this discussion advocates for there not being a default at all. These file formats have quite different characteristics and users should probably make a conscious choice about them.

Reply:
In pyarrow we default to parquet but perhaps that is for legacy reasons.  I'm fine with no default here.  However, we should then get rid of the implicit no-arg constructor and add a single-argument constructor taking in write options since this will now have a required argument.  E.g.

```
FileSystemDatasetWriteOptions(std::shared_ptr<FileWriteOptions> file_write_options) : file_write_options(std::move(file_write_options)) {}
```